packages/mobx-state-tree/__tests__/core/optional-extension.test.ts

```diff
@@ -0,0 +1,264 @@
+import { getSnapshot, types, unprotect } from "../../src"
+
+describe("null as default", () => {
+    describe("basic tests", () => {
+        const M = types.model({
```

Comment:
test suggestion, with `[null]`, `x: undefined` or `y` being omitted should give type errors

Reply:
they do (tested manually), but how to test typings automatically? :-/

Reply:
I think I know now what you mean. added

```ts
        const M = types.model({
            a: types.optional(types.number, 5, [null]),
            b: types.optional(types.number, 6, [null])
        })

        {
            const m = M.create({
                a: null,
                b: null
            })
            expect(m.a).toBe(5)
            expect(m.b).toBe(6)
        }

        if (process.env.NODE_ENV !== "production") {
            expect(() => {
                M.create({
                    a: null,
                    b: undefined as any // undefined is not valid
                })
            }).toThrowError("value `undefined` is not assignable to type: `number`")

            expect(() => {
                M.create({
                    a: null
                    // b: null missing, but should be there
                } as any)
            }).toThrowError("value `undefined` is not assignable to type: `number`")
        }
```